memory/player_party_manager.py

```diff
@@ -46,10 +52,50 @@ def update(self: Self) -> None:
                     self._read_gameobject_position()
                     self._read_movement_state()
                     self._read_leader_character()
+                    self._read_current_party()
+
             except Exception as _e:
                 # logger.debug(f"PlayerPartyManager Reloading {type(_e)}")
                 self.__init__()
 
+    def _read_current_party(self: Self) -> [PlayerPartyCharacter]:
+        try:
+            current_party_ptr = self.memory.follow_pointer(self.base, [0x98, 0x0])
+        except Exception:
+            self.players = []
```

Comment:
There seems to be a mix and match between `self.players` and `self.current_party` here, with `self.current_party` seeming to be the "real" one. References to `self.players` should be removed.

Additionally, `self.current_party` should be assigned a default value and type in the `__init__`.